src/main/java/Duke.java

```diff
@@ -1,10 +1,236 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    /**
+     * Sets up the required objects, loads up the data from the storage file.
+
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.loadtxt());
+        } catch (Exception e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+    /**
+     * Prints the welcome message,and scan user input.
+     */
+    public void run() {
+        ui.showWelcome();
+        Scanner sc = new Scanner(System.in);
+        while(sc.hasNextLine()){
+            String s = sc.nextLine();
+            String[] a = Parser.parse(s);
+            switch (a[0]) {
+                case "bye" :
+                    ui.bye();
+                    return;
+                case "list" :
+                    list(tasks.getlist());
+                    break;
+                case "done":
+                    int n = Integer.parseInt(a[1]);
+                    done(tasks.getlist(), n-1);
+                    break;
+                case "todo":
+                    try{
+                        todo(tasks.getlist(),s);
+                    }
+                    catch (ErrorException e) {
```

Comment:
Nice use of switch, maybe can also move the try catch block outside the switch cases, since you're always catching `ErrorException`.